components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ThrottlingApiServiceImpl.java

```diff
@@ -892,14 +892,11 @@ public Response throttlingBlacklistPost(BlockingConditionDTO body, String conten
             if (RestApiUtil.isDueToResourceAlreadyExists(e)) {
                 RestApiUtil.handleResourceAlreadyExistsError("A black list item with type: " + body.getConditionType() + ", value: " + body.getConditionValue() + " already exists", e, log);
             } else {
-                String errorMessage =
-                        "Error while adding Blocking Condition. Condition type: " + body.getConditionType() + ", " +
-                                "value: " + body.getConditionValue();
+                String errorMessage = "Error while adding Blocking Condition : " + e.getMessage();
                 RestApiUtil.handleInternalServerError(errorMessage, e, log);
             }
         } catch (URISyntaxException | ParseException e) {
-            String errorMessage =
-                    "Error while retrieving Blocking Condition resource location. Condition type: " + body.getConditionType() + ", value: " + body.getConditionValue();
+            String errorMessage = "Error while retrieving Blocking Condition resource location: "  + e.getMessage();
```

Comment:
Add block condition context info so that someone can correlate the error messages with corresponding block condition.